### PR TITLE
[KYUUBI #3300] [FOLLOWUP] simply reflection method callings in Authz's AccessRequest

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

To simplify the method calling via reflection in Authz's `AccessRequest`, including codes from pull #3308 introduced.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by bowenliang123** [comment]
cc @yaooqinn .

Plus, it  introduced `invokeWithCast` in AuthZUtils for simplifying object casting and throwing readable exception message when the method is not found. It also improves readability in more Scala style by bringing the type declaration of return value closer to the result variable.

e.g. from 
```
val name = invoke(object, "method").asInstanceOf[String]
```
to 
```
val name = invokeWithCast[String](object, "method")
```

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/AuthZUtils.scala [hunk 1]
```diff
@@ -68,6 +68,20 @@ private[authz] object AuthZUtils {
     method.invoke(obj, values: _*)
   }
 
+  def invokeWithCast[T](
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/AuthZUtils.scala:71]: we can name it invokeAs

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/AuthZUtils.scala]: Good, `invokeAs` is a better one.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala [hunk 2]
```diff
@@ -45,15 +45,12 @@ object AccessRequest {
     req.setUserGroups(userGroups)
     req.setAction(opType.toString)
     try {
-      val getRoles = SparkRangerAdminPlugin.getClass.getMethod(
+      val roles = invokeAs[java.util.Set[String]](
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala:48]: pay attention to the error handling

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala [hunk 3]
```diff
@@ -63,12 +60,8 @@ object AccessRequest {
       case _ => req.setAccessType(accessType.toString.toLowerCase)
     }
     try {
-      val getClusterName = SparkRangerAdminPlugin.getClass.getMethod("getClusterName")
-      getClusterName.setAccessible(true)
-      val clusterName = getClusterName.invoke(SparkRangerAdminPlugin)
-      val setClusterName = req.getClass.getMethod("setClusterName", classOf[String])
-      setClusterName.setAccessible(true)
-      setClusterName.invoke(req, clusterName)
+      val clusterName = invokeAs[String](SparkRangerAdminPlugin, "getClusterName")
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala:63]: ditto

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala [hunk 4]
```diff
@@ -45,32 +45,25 @@ object AccessRequest {
     req.setUserGroups(userGroups)
     req.setAction(opType.toString)
     try {
-      val getRoles = SparkRangerAdminPlugin.getClass.getMethod(
+      val roles = invokeAs[java.util.Set[String]](
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala:48]: how about using `import java.util.{Set => JSet}`

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala [hunk 5]
```diff
@@ -81,23 +74,11 @@ object AccessRequest {
 
   private def getUserGroupsFromUserStore(user: UserGroupInformation): Option[util.Set[String]] = {
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala:75]: here we use util.Set, how about bringing into correspondence with `java.util.{Set => JSet}`

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala [hunk 6]
```diff
@@ -81,23 +74,11 @@ object AccessRequest {
...
-
-      val userGroupMappingMap: mutable.Map[String, util.Set[String]] =
-        mapAsScalaMap(getUserGroupMapping.invoke(userStore)
-          .asInstanceOf[util.HashMap[String, util.Set[String]]])
-
-      userGroupMappingMap.get(user.getShortUserName)
+      val storeEnricher = invoke(SparkRangerAdminPlugin, "getUserStoreEnricher")
+      val userStore = invoke(storeEnricher, "getRangerUserStore")
+      val userGroupMapping =
+        invokeAs[util.HashMap[String, util.Set[String]]](userStore, "getUserGroupMapping")
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala:80]: ditto

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/AuthZUtils.scala [hunk 7]
```diff
@@ -68,6 +68,20 @@ private[authz] object AuthZUtils {
...
   }
 
+  def invokeAs[T](
+      obj: AnyRef,
+      methodName: String,
+      args: (Class[_], AnyRef)*): T = {
+    Try {
+      invoke(obj, methodName, args: _*)
+    } match {
+      case Success(value) => value.asInstanceOf[T]
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/AuthZUtils.scala:78]: should be better to add these in invoke

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/AuthZUtils.scala]: ok,  exception throwing moved to the `invoke` method.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala (hunk 5 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala]: addressed.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala (hunk 4 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala]: addressed.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala (hunk 3 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala:63]: addressed.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala (hunk 2 above)
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala]: addressed.

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**